### ISSUE TITLE
Do not silently fall back on pure-Python when C code missing

### ISSUE THREAD

**Issue by peterjc** [biopython/biopython#657]
See discussion on #656 / #656 where importing the C-accelerated PWM code was failing under Python 3, with ``Bio.Motif`` silently falling back on the pure-Python implementation.

This can also happen with other areas of the code such as ``Bio.cpairwise2``.

This behaviour is expected and desirable under other Python implementations like PyPy and Jython, but under C Python should probably trigger a warning or simply raise an exception?

**Comment by mruffalo** [comment]
Personally, I'd rather get an exception than a warning -- there's a decent chance that I'd miss a warning when submitting jobs to a cluster.

It looks like Python 2.6 is the minimum version supported by BioPython, so something like this seems straightforward:

```
import platform

try:
    import _c_version_of_whatever
except ImportError:
    if platform.python_implementation() == 'CPython':
        raise
else:
    # load/activate Python fallback version
```

**Comment by peterjc** [comment]
That sounds good to me - @mdehoon?

We need something like this in at least ``Bio/pairwise2.py``, ``Bio/motifs/matrix.py`` (perhaps done once at top level rather than inside the method as you noted earlier), and ``Bio/Nexus/Nexus.py``.

Looking over ``setup.py``, I think all the other bits of C code don't have a pure-Python fallback.

**Comment by peterjc** [comment]
Regarding the code snippet, I think else clause should be indented to be part of the ``ImportError`` handler? If some other strange exception was triggered by the import line, we don't want to hide that.

**Comment by mruffalo** [comment]
Good catch, thanks. Side effect of writing that quickly for that comment :)

I'll make the appropriate tweaks to `Bio/motifs/matrix.py` and should be able to submit a PR today.

**Comment by mdehoon** [comment]
Fixed by #659 , thanks!
